src/eventer/eventer_SSL_fd_opset.c

```diff
@@ -703,7 +780,37 @@ eventer_ssl_ctx_t *
 eventer_ssl_ctx_new(eventer_ssl_orientation_t type,
                     const char *layer,
                     const char *certificate, const char *key,
-                    const char *ca, const char *ciphers) {
+                    const char *ca_chain, const char *ciphers) {
+  mtev_hash_table settings;
+  mtev_hash_init(&settings);
+  mtev_hash_store(&settings, "layer", strlen("layer"), layer);
+  mtev_hash_store(&settings, "certificate", strlen("certificate"), certificate);
+  mtev_hash_store(&settings, "key", strlen("key"), key);
+  mtev_hash_store(&settings, "ca_chain", strlen("ca_chain"), ca_chain);
+  mtev_hash_store(&settings, "ciphers", strlen("ciphers"), ciphers);
+  eventer_ssl_ctx_t *ctx = eventer_ssl_ctx_new_ex(type, &settings);
+  mtev_hash_destroy(&settings,NULL,NULL);
+  return ctx;
+}
+
+eventer_ssl_ctx_t *
+eventer_ssl_ctx_new_ex(eventer_ssl_orientation_t type,
+                       mtev_hash_table *settings) {
+  const char *layer = mtev_hash_dict_get(settings, "layer");
+  const char *certificate = mtev_hash_dict_get(settings, "certificate");
+  if(!certificate) certificate = mtev_hash_dict_get(settings, "certificate_file");
+  const char *key = mtev_hash_dict_get(settings, "key");
+  if(!key) key = mtev_hash_dict_get(settings, "key_file");
+  const char *ca = mtev_hash_dict_get(settings, "ca_chain");
+  if(!ca) mtev_hash_dict_get(settings, "ca_file");
+  const char *ciphers = mtev_hash_dict_get(settings, "ciphers");
+  const char *ca_accept = mtev_hash_dict_get(settings, "ca_accept");
+  if(!ca_accept) ca_accept = ca;
+  const char *crl = mtev_hash_dict_get(settings, "crl");
+  const char *npn = mtev_hash_dict_get(settings, "alpn");
+  if(!npn) npn = mtev_hash_dict_get(settings, "npn");
+  if(!npn) npn="h2,http/1.1";
+
```

Comment:
is there any logging to show what parameters mtev read in - that is important in order for a user to confirm that mtev took/parsed the correct settings into itself and avoid chasing your tail for hours

Reply:
I'll add.

Reply:
actually when set, it will print to ssldb deeper down.

Reply:
ok cool